docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,208 @@
+# cuDF Architecture
+
+The cuDF library is a GPU-accelerated, [Pandas-like](https://pandas.pydata.org/) DataFrame library.
+pandas APIs provide users a greate deal of power and flexibility, and we aim to match that.
+As a result, a key design challenge for cuDF is finding the simplest, most performant approaches to mimic pandas APIs.
+
+At a high level, cuDF is structured in three layers, each of which serves a distinct purpose in this regard:
+
+1. The Frame layer: The user-facing implementation of pandas-like data structures.
```

Comment:
Does this definition of the "Frame layer" include other public types like `GroupBy` or our custom data types?

Reply:
Yes, although I had not intended to mention to many of these classes since I don't think this document needs to be quite that comprehensive. With my current edits I've added some additional information about these though.